pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -964,9 +964,15 @@ type FilesystemSpec struct {
 	// +nullable
 	MetadataPool PoolSpec `json:"metadataPool"`
 
-	// The data pool settings
-	// +nullable
-	DataPools []PoolSpec `json:"dataPools"`
+	// The data pool settings, deprecated.
+	// At least DataPools or DataPoolsMapped should be specified
```

Comment:
I assume it is not supported for users to specify both of these?

```suggestion
	// (Deprecated in favor of dataPoolsMapped) The data pool settings.
	// Exactly one of DataPools or DataPoolsMapped (preferred) must be specified.
```

Reply:
It's discussable. TBH, give a user an ability to specify both at time will not lead to any problems. At least, we may check that pools specified in map are not intersect with pools from array. Wdyt?